### PR TITLE
RetroPlayer: GL(ES) don't hardcode pixelformats

### PR DESCRIPTION
**Pull request by lrusak**
Before we were hardcoding in the pixelformat. Now we can use the pixel format as given by libretro

This PR also activates the `CRPRendererOpenGL` and `CRPRendererOpenGLES` for `X11` and `GBM` respectively. Other platforms can be added after they have been tested.

build and run tested on Linux using `X11` and `GBM` for `GL` and `GLES` respectively.

### CODE REVIEW

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

File: xbmc/rendering/RenderSystem.h
Comment (FernetMenta) [inline xbmc/rendering/RenderSystem.h]: this is platform specific and should be handled at a lower level

File: xbmc/rendering/RenderSystem.h
Comment (FernetMenta) [inline xbmc/rendering/RenderSystem.h]: demote to RenderSystemGLES

File: xbmc/rendering/gles/RenderSystemGLES.cpp
Comment (FernetMenta) [inline xbmc/rendering/gles/RenderSystemGLES.cpp]: imo those flags are crap in general and adding an additional one makes zero sense. instead of querying the flag you can just query for the extension.

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGL.h
Comment (Rechi) [inline xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPRendererOpenGL.h]: override missing

**Comment by garbear** [comment]
Looks like the last commit caused some errors. Can you enable GL/GLES on all platforms?

**Review by FernetMenta — COMMENTED** [review]
where is your VAO?

**Comment by garbear** [comment]
Now that we're getting to the final review stages, before merge can you squash the following commits?

* "Retroplayer: add extra rendering functions to CRenderContext"
* "RPRendererOpenGL: add override for UploadTexture method"
* "RetroPlayer: Move GL configuration to ConfigureInternal()"
* "RPRenderingOpenGLES: don't hardcode pixelformats"
* (possibly) "RPRendererOpenGLES: use VBOs"

The purpose is to feature-delineate commits, especially when a commit for a feature is affected by code of a later commit (my refactor one). Combining different platforms into one feature commit is also helpful, because devs are (hopefully) only troubleshooting one platform at a time.

For example, you can see in the [RetroPlayer PR](https://github.com/xbmc/xbmc/pull/11022/commits) that all commits are feature-delineated.

**Comment by lrusak** [comment]
@FernetMenta VAO's are only supported by GLES 3.0 https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glGenVertexArrays.xhtml

**Review by garbear — APPROVED** [review]

**Comment by FernetMenta** [comment]
@lrusak you might run into compatibility issues when using VBO without VAO.

**Pull request merged**
